cypress/e2e/requests/cancel-request.cy.js

```diff
@@ -0,0 +1,184 @@
+  let originalCirculationRules;
+  let itemData;
+  const instanceData = {
+    title: `Instance title_${getRandomPostfix()}`,
+    itemBarcode: `item${generateUniqueItemBarcodeWithShift()}`,
+  };
+  const testData = {
+    folioInstances: InventoryInstances.generateFolioInstances(),
+    servicePoint: ServicePoints.getDefaultServicePointWithPickUpLocation(),
+    requestsId: '',
+  };
+  const requestPolicyBody = {
+    requestTypes: [REQUEST_TYPES.PAGE],
+    name: `requestPolicy${getRandomPostfix()}`,
+    id: uuid(),
+  };
+
+  before('Create test data', () => {
+    cy.getAdminToken().then(() => {
+      ServicePoints.createViaApi(testData.servicePoint);
+      testData.defaultLocation = Location.getDefaultLocation(testData.servicePoint.id);
+      Locations.createViaApi(testData.defaultLocation).then((location) => {
+        InventoryInstances.createFolioInstancesViaApi({
+          folioInstances: testData.folioInstances,
+          location,
+        });
+        instanceData.instanceId = testData.folioInstances[0].instanceId;
+
+        itemData = testData.folioInstances[0];
+      });
+      RequestPolicy.createViaApi(requestPolicyBody);
+      CirculationRules.getViaApi().then((circulationRule) => {
+        originalCirculationRules = circulationRule.rulesAsText;
+        const ruleProps = CirculationRules.getRuleProps(circulationRule.rulesAsText);
+        ruleProps.r = requestPolicyBody.id;
+        addedCirculationRule =
+          't ' +
+          testData.loanTypeId +
+          ': i ' +
+          ruleProps.i +
+          ' l ' +
+          ruleProps.l +
+          ' r ' +
+          ruleProps.r +
+          ' o ' +
+          ruleProps.o +
+          ' n ' +
+          ruleProps.n;
+        CirculationRules.addRuleViaApi(
+          originalCirculationRules,
+          ruleProps,
+          't ',
+          testData.loanTypeId,
+        );
+      });
+      cy.createTempUser([Permissions.requestsAll.gui, Permissions.checkinAll.gui])
+        .then((userProperties) => {
+          testData.user = userProperties;
+        })
+        .then(() => {
+          UserEdit.addServicePointsViaApi(
+            [testData.servicePoint.id],
+            testData.user.userId,
+            testData.servicePoint.id,
+          );
+          cy.getInstance({
+            limit: 1,
+            expandAll: true,
+            query: `"id"=="${instanceData.instanceId}"`,
+          }).then((instance) => {
+            instanceData.instanceHRID = instance.hrid;
+            cy.login(testData.user.username, testData.user.password, {
+              path: TopMenu.requestsPath,
+              waiter: Requests.waitLoading,
+            });
+            NewRequest.openNewRequestPane();
+            NewRequest.waitLoadingNewRequestPage();
+            NewRequest.enterHridInfo(instanceData.instanceHRID);
+            NewRequest.verifyHridInformation([instanceData.title]);
+            NewRequest.enterRequesterInfoWithRequestType(
```

Comment:
why don't you just use api method for creating request if this steps in preconditions?
```
Requests.createNewRequestViaApi({
  fulfillmentPreference: FULFILMENT_PREFERENCES.HOLD_SHELF,
  instanceId: testData.instanceId,
  pickupServicePointId: testData.userServicePoint.id,
  requestDate: new Date(),
  requestLevel: REQUEST_LEVELS.TITLE,
  requestType: REQUEST_TYPES.PAGE,
  requesterId: testData.user.userId,
}).then((createdRequest) => {
  requestId = createdRequest.body.id;
});
```